ai-ml/trainium-inferentia/examples/ray-serve/stable-diffusion-inf2/ray_serve_stablediffusion.py

```diff
@@ -0,0 +1,55 @@
+from io import BytesIO
+from fastapi import FastAPI
+from fastapi.responses import Response
+import os
+import base64
+
+from ray import serve
+
+app = FastAPI()
+
+neuron_cores = 2
+
+@serve.deployment(num_replicas=1, route_prefix="/")
+@serve.ingress(app)
+class APIIngress:
+    def __init__(self, diffusion_model_handle) -> None:
+        self.handle = diffusion_model_handle
+
+    @app.get(
+        "/imagine",
+        responses={200: {"content": {"image/png": {}}}},
+        response_class=Response,
+    )
+    async def generate(self, prompt: str):
+
+        image_ref = await self.handle.generate.remote(prompt)
+        image = await image_ref
+        file_stream = BytesIO()
+        image.save(file_stream, "PNG")
+        return Response(content=file_stream.getvalue(), media_type="image/png")
+
+
+@serve.deployment(
+    ray_actor_options={
+        "resources": {"neuron_cores": neuron_cores},
+        "runtime_env": {"env_vars": {"NEURON_CC_FLAGS": "-O1"}},
+    },
+    autoscaling_config={"min_replicas": 1, "max_replicas": 1},
```

Comment:
this with the `neuron_cores` makes me believe we are consuming only 2 neuron cores but yet we're requesting 12 neuron cores (6 devices, 2 cores per device) - what resources requirements does this model/pattern require?

Reply:
For StableDiffusion, allocating more than 2 cores may not improve the performance. I didn't find any documentation that provides guidance on this. I've tested with 2 cores on Inf2.24xl. Will update the values accordingly.